transport/interface.go

```diff
@@ -23,15 +23,21 @@ package transport
 import (
 	"time"
 
+	"github.com/opentracing/opentracing-go"
 	"golang.org/x/net/context"
 )
 
 // Request is the fields used to make an RPC.
 type Request struct {
-	Method  string
-	Timeout time.Duration
-	Headers map[string]string
-	Body    []byte
+	Method          string
+	Timeout         time.Duration
+	Headers         map[string]string
+	Baggage         map[string]string
+	RoutingKey      string
+	RoutingDelegate string
+	ShardKey        string
+	Body            []byte
+	Tracer          opentracing.Tracer
```

Comment:
The `Request` and `Response` objects only contain values that represent the on-wire presentation currently. I'm not sure if Tracer should be here, should we move it into the `Call` interface instead?